lib/appraisal/bundler_dsl.rb

```diff
@@ -34,6 +34,10 @@ def group(*names, &block)
       @groups[names].run(&block)
     end
 
+    def install_if(*args)
+      raise "Put some code in here"
```

Comment:
This is a bit of a fun one, as it's a method call which takes a lambda. I think we should evaluate that and then delegate to the existing `gem` command:

```ruby
def gem(name, *requirements)
  @dependencies.add(name, substitute_git_source(requirements))
end
```

These might help:

* https://github.com/bundler/bundler/blob/6a3b09b2292405cab6b5f4d63004b5c8b782b469/lib/bundler/dsl.rb#L237
* https://github.com/bundler/bundler/blob/3032f4aa447ffa58bfc2fa706472591deb696a66/spec/install/gemfile/install_if.rb#L3

Reply:
Thanks for the feedback! I think evaluating the lambda as this point would be aggressive, as really its about being done when the gemfile is hit. It's used for ENV checks and all sorts of things, as well as ruby checks. 

We dont want the lambda evaluated on generate, we want it evaluated on bundle install. No idea how to do that and im too pneumonia sick to think :D

Reply:
👍 . Were you able to look at this one again? (I hope you're all better now!)

Reply:
Back in good health, but I'm still not sure how to do this and tbh I no longer need it. Happy to close, but I think it'd be a good feature for your project.